go/oasis-test-runner/cmd/cmp/cmp.go

```diff
@@ -0,0 +1,531 @@
+	if len(tests) == 0 {
+		for _, s := range common.DefaultScenarios {
+			tests = append(tests, s.Name())
+		}
+	}
+	succ := true
+	for _, test := range tests {
+		labels := map[string]string{}
+		if viper.IsSet(cfgMetricsSourceGitBranch) {
+			labels[oasis.MetricsLabelGitBranch] = viper.GetString(cfgMetricsSourceGitBranch)
+		}
+
+		// Query parameter value-specific tests only.
+		s := common.Scenarios[test]
+		if s != nil {
+			common.Scenarios[test].Parameters().VisitAll(func(f *flag.Flag) {
+				param := fmt.Sprintf(common.ScenarioParamsMask, test, f.Name)
+				if viper.IsSet(param) {
+					// TODO: We should support all parameter set combinations in the future like we do in oasis-test-runner.
+					labels[metrics.EscapeLabelCharacters(f.Name)] = viper.GetStringSlice(param)[0]
+				}
+			})
+		}
+
+		// Set other required Prometheus labels, if passed.
+		for k, v := range metrics.GetStringMapString(viper.GetString(metrics.CfgMetricsLabels)) {
+			labels[k] = v
+		}
+
+		sInstances, err := getCoarseBenchmarkInstances(ctx, test, labels)
+		if err != nil {
+			cmpLogger.Error("error querying for source test instances", "err", err)
+			os.Exit(1)
+		}
+		sNames := instanceNames(sInstances)
+		tInstances, err := getCoarseBenchmarkInstances(ctx, test, labels)
+		if err != nil {
+			cmpLogger.Error("error querying for target test instances", "err", err)
+			os.Exit(1)
+		}
+		tNames := instanceNames(tInstances)
+
+		if len(sNames) == 0 {
+			cmpLogger.Info("test does not have any source benchmark instances to compare, ignoring", "test", test)
+			continue
+		}
+		if len(tNames) == 0 {
+			cmpLogger.Info("test does not have any target benchmark instances to compare, ignoring", "test", test)
+			continue
+		}
+
+		var sInstance, tInstance *model.SampleStream
+		if sNames[len(sNames)-1] != tNames[len(tNames)-1] {
+			// Benchmark instances differ e.g. because of different gitBranch.
+			sInstance = sInstances[len(sInstances)-1]
+			tInstance = tInstances[len(tInstances)-1]
+		} else {
+			// Last benchmark instances are equal, pick the pre-last one from the target instances.
+			if len(tNames) < 2 {
+				cmpLogger.Info("test has only one benchmark instance, ignoring", "test", test, "source_instances", sNames, "target_instances", tNames)
+				continue
+			}
+			sInstance = sInstances[len(sInstances)-1]
+			tInstance = tInstances[len(tInstances)-2]
+		}
+		cmpLogger.Info("obtained source and target instance", "test", test, "source_instance", instanceName(sInstance), "target_instance", instanceName(tInstance))
+
+		for _, m := range userMetrics {
+			// Don't put succ = succ && f oneliner here, because f won't get executed once succ = false.
+			fSucc := fetchAndCompare(ctx, m, test, sInstance, tInstance)
+			succ = succ && fSucc
+		}
+	}
+
+	if !succ {
+		os.Exit(1)
+	}
+
+	defer cancel()
+}
+
+// Register oasis-test-runner cmp sub-command and all of it's children.
+func RegisterCmpCmd(parentCmd *cobra.Command) {
```

Comment:
This will be called as `cmp.RegisterCmpCmd` so no need for two cmps, use just `Register` to be consistent with other parts of the code.